ykllvmwrap/src/ykllvmwrap.cc

```diff
@@ -353,16 +355,24 @@ extern "C" void *__ykllvmwrap_irtrace_compile(char *FuncNames[], size_t BBs[],
           // FIXME Deal with calls we cannot or don't want to inline.
           if (Tracing) {
             inlined_calls.push_back(CI);
+            auto PrevVMap = vmaps.back();
+            vmaps.push_back(new llvm::ValueToValueMapTy());
+            VMap = vmaps.back();
+            // Copy over globals
+            auto OutermostVMap = vmaps.front();
```

Comment:
Does this assume that all globals are pulled in my the first frame?

Reply:
Define "pulled in"? I'm adding all globals to the first frame, because we need to be able to find them once the trace is done in order to initialise them.

Reply:
So that we don't have to iterate all VMaps collecting a set? Ah, I see. I think a comment is required here.